conda-store-server/conda_store_server/worker/tasks.py

```diff
@@ -43,6 +44,17 @@ def worker(self):
 
             self._worker.initialize()
 
+        # Installs a signal handler that terminates the process on Ctrl-C
+        # (SIGINT)
+        # Note: the following would not be enough to terminate beat and other
+        # tasks, which is why the code below explicitly calls sys.exit
+        # > from celery.apps.worker import install_worker_term_hard_handler
+        # > install_worker_term_hard_handler(self._worker, sig='SIGINT')
+        def _shutdown(*args, **kwargs):
+            return sys.exit(1)
```

Comment:
should this be exit value of 1? A requested exit that exits cleanly should be a zero exit status (the default in this case)

Reply:
It *should* be non-zero, yes. Despite being user-requested, it's still abnormal termination since tasks are immediately terminated mid-execution, which can have side-effects (like partially downloaded files, etc.).

The reason I terminate tasks immediately is because some of them might take a very long time to terminate cleanly and that use case is already [handled by SIGTERM in Celery](https://docs.celeryq.dev/en/stable/userguide/workers.html#stopping-the-worker) anyway. Waiting for tasks to terminate is not something I want on Ctrl-C.

Because of how we start Celery via multiprocessing and Celery additionally spawning processes on its own, the current mechanism is the only way I've found that doesn't leave stale processes.

Going back to the exit code, this is up to the application to define. I use `exit 1` because it's a good default as we don't have a map of exit codes (I don't think it's very important for conda-store because it's not a script that will be used in shell pipelines).

For example, `ls` has these exit statuses (via `man ls`):

```
   Exit status:
       0      if OK,
       1      if minor problems (e.g., cannot access subdirectory),
       2      if serious trouble (e.g., cannot access command-line argument).
```

Another example, `conda` returns 1 on Ctrl-C:
```
$ conda install constructor
Retrieving notices: ...working... done
^C
CondaError: KeyboardInterrupt

$ echo $?
1
```